src/Nethermind/Nethermind.Network/P2P/Subprotocols/Les/StatusMessage.cs

```diff
@@ -0,0 +1,84 @@
+{
+    public class StatusMessage : P2PMessage
+    {
+
+        public static class KeyNames
+        {
+            public const string ProtocolVersion = "protocolVersion";
+            public const string ChainId = "networkId";
+            public const string TotalDifficulty = "headTd";
+            public const string BestHash = "headHash";
+            public const string HeadBlockNo = "headNum";
+            public const string GenesisHash = "genesisHash";
+            public const string AnnounceType = "announceType";
+            public const string ServeHeaders = "serveHeaders";
+            public const string ServeChainSince = "serveChainSince";
+            public const string ServeRecentChain = "serveRecentChain";
+            public const string ServeStateSince = "serveStateSince";
+            public const string ServeRecentState = "serveRecentState";
+            public const string TxRelay = "txRelay";
+            public const string BufferLimit = "flowControl/BL";
+            public const string MaximumRechargeRate = "flowControl/MRR";
+            public const string MaximumRequestCosts = "flowControl/MRC";
+        }
+
+        public override int PacketType { get; } = LesMessageCode.Status;
+        public override string Protocol => "les";
+        public byte ProtocolVersion { get; set; }
+        public UInt256 ChainId { get; set; }
+        public UInt256 TotalDifficulty { get; set; }
+        public Keccak BestHash { get; set; }
+        public long HeadBlockNo { get; set; }
+        public Keccak GenesisHash { get; set; }
+        #region optional
+        public byte? AnnounceType { get; set; } // sent from client only
+        public bool ServeHeaders { get; set; }
+        public long? ServeChainSince { get; set; }
+        public long? ServeRecentChain { get; set; }
+        public long? ServeStateSince { get; set; }
+        public long? ServeRecentState { get; set; }
+        public bool TxRelay { get; set; }
+        public int? BufferLimit { get; set; }
+        public int? MaximumRechargeRate { get; set; }
+        // These are the initial defaults from geth.
+        // It probably doesn't make sense to define them here in the finished implementation, since the client will want to use the values supplied by the server
+
+        // TODO: Benchmark finished implementation and update based on our actual serve times.
+        // TODO: Implement cost scaling to account for users with different capabilities - https://github.com/ethereum/go-ethereum/blob/01d92531ee0993c0e6e5efe877a1242bfd808626/les/costtracker.go#L437
+        // TODO: Implement multiple cost lists, so it can be limited based on the minimum of available bandwidth, cpu time, etc. - https://github.com/ethereum/go-ethereum/blob/01d92531ee0993c0e6e5efe877a1242bfd808626/les/costtracker.go#L186
+        // TODO: This might be better as a dictionary
+        public RequestCostItem[] MaximumRequestCosts = new RequestCostItem[]
```

Comment:
this should be static

Reply:
Will do. My thinking at the time, was that we might want to have a different table per client, maybe based on reputation, or maybe we'd give a way to provide a list of clients you own, so that their requests have zero cost.

Though, thinking about it now, that would probably make more sense as changes to the recharge rate and/or buffer limit.

Reply:
FWIW I love the idea of being able to specify a list of light clients I own/trust to allow un-throttled queries, etc.

Reply:
Cool! It seems useful to me also. The problem @tkstanczak  is pointing out with the way it's currently defined is that the list is duplicated for every message, taking up more memory space than necessary. 

We should be able to do both though - I think eventually, this will probably end up referring to a static copy defined elsewhere, alongside the rest of the cost tracking logic.